### PR TITLE
Setup server for new specifications

### PR DESCRIPTION
**Pull request by RobertWSaunders**
## Proposed Change

This sets up the server such that it can satisfy the requirements our clients have via GraphQL and a relational database. This purges code that is coupled to MongoDB (and as a result Mongoose) in favour of utilizing a new relational data model which this PR sets up the basic config to enable developers to build the data layer and interface (via GraphQL).

### How did you do this?

Purging a lot of code and setting up Sequelize and Apollo Server.

### Why are you choosing this approach?

Make our platform better and enable developers to work with new tools!

### Any open questions you want to ask code reviewers?

Nope!

### List of changes:

  - Purge old REST API
  - Purge all tests coupled to REST API
  - Purge Mongoose schemas and models
  - Add Sequelize config
  - Add GraphQL config
  - Purge Google Cloud Platform deployment info

## Pull Request Checklist:

  - [x] My submission passes all tests.
  - [x] I have lint my code locally prior to submission.
  - [x] I have written new tests for my core changes (where applicable).
  - [x] I have referenced all useful information (issues, tasks, etc) in the pull request.

### CODE REVIEW

File: packages/server/db/models/oauth-client.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      }
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: true
+      }
+    },
+    host: {
```
Comment (RobertWSaunders) [inline packages/server/db/models/oauth-client.js]: I added this field in!

File: packages/server/db/models/oauth-client.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const OAuthClient = sequelize.define("OAuthClient", {
+    id: {
+      primaryKey: true,
+      type: DataTypes.STRING,
+      defaultValue: DataTypes.UUIDV4,
+      validate: {
+        isUUID: 4
+      }
+    },
+    name: {
```
Comment (RobertWSaunders) [inline packages/server/db/models/oauth-client.js]: I added this field in!

File: packages/server/db/models/oauth-client.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    host: {
+      type: DataTypes.STRING,
+      allowNull: true,
+      validate: {
+        notEmpty: true
+      }
+    },
+    clientSecret: {
+      type: DataTypes.STRING,
+      allowNull: true,
```
Comment (RobertWSaunders) [inline packages/server/db/models/oauth-client.js]: Make this nullable!

File: packages/server/db/models/user.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+          isEmail: true,
+          isLowercase: true,
+          set(val) {
+            this.setDataValue("email", val.toLowerCase());
+          }
+        }
+      },
+      phoneNumber: {
+        type: DataTypes.STRING,
+        allowNull: true,
```
Comment (RobertWSaunders) [inline packages/server/db/models/user.js]: Made this nullable!

File: packages/server/db/models/user.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        validate: {
+          notEmpty: true,
+          is: PHONE_NUMBER_REGEX
+        }
+      },
+      dateOfBirth: {
+        type: DataTypes.DATE,
+        allowNull: true,
```
Comment (RobertWSaunders) [inline packages/server/db/models/user.js]: Made this nullable!

File: packages/server/db/seeders/20181124185518-seed-data.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+const faker = require("faker");
+const bcrypt = require("bcrypt");
+const uuidv4 = require("uuid/v4");
+
+module.exports = {
```
Comment (RobertWSaunders) [inline packages/server/db/seeders/20181124185518-seed-data.js]: I hate this shit but its a work in progress!

File: packages/server/gql/definitions/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+const { gql } = require("apollo-server-express");
+const fs = require("fs");
+
+const GRAPHQL_FILE_REGEX = /.+\.graphql$/;
+
+const directories = ["schema", "enums", "interfaces", "scalars", "types"];
+
+let schemaString = "";
+
+directories.forEach((directory) => {
```
Comment (RobertWSaunders) [inline packages/server/gql/definitions/index.js]: Definitely a better way to write this!

File: packages/server/gql/resolvers/scalars/Date.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+const { GraphQLScalarType, Kind } = require("graphql");
+
+module.exports = {
```
Comment (RobertWSaunders) [inline packages/server/gql/resolvers/scalars/Date.js]: Still need to implement these custom scalars.

File: packages/server/package.json [hunk 9]
```diff
@@ -4,42 +4,50 @@
   "description": "The server for the QHacks Dashboard!",
   "main": "index.js",
   "scripts": {
-    "dev": "npm run build && NODE_ENV=development nodemon index.js",
+    "dev": "npm run build && NODE_ENV=development nodemon --ext graphql,js,json index.js",
```
Comment (RobertWSaunders) [inline packages/server/package.json]: Added this so that `.graphql` files are watched!

File: packages/server/db/models/oauth-client.js (hunk 2 above)
Comment (joeytepp) [reply packages/server/db/models/oauth-client.js]: Nice 👌

File: packages/server/db/models/oauth-client.js (hunk 1 above)
Comment (joeytepp) [reply packages/server/db/models/oauth-client.js]: Sounds good 👍

**Pull request merged**